include/aws/s3/s3_client.h

```diff
@@ -689,6 +689,15 @@ struct aws_s3_meta_request_options {
      * from the buffer and compare them them to previously uploaded part checksums.
      */
     struct aws_s3_meta_request_resume_token *resume_token;
+
+    /*
+     * Optional.
+     * Object size hint in bytes.
+     * This hint helps optimize downloads for small (less than or equal to part_size) or large files.
```

Comment:
trivial: me trying to help people understand what "optimize" means.
```suggestion
     * The optimal strategy for downloading a file depends on its size.
     * Set this hint to help the S3 client choose the best strategy for this particular file.
```